### PR TITLE
Fixing  extraction error because of long path

### PR DESCRIPTION
**Pull request by anuragc617**
Fixing #10879

### CODE REVIEW

File: Tasks/UseDotNetV2/package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
   "dependencies": {
     "@types/mocha": "^5.2.6",
     "@types/semver": "6.0.0",
+    "@types/uuid": "^3.4.5",
```
Comment (bishal-pdMSFT) [inline Tasks/UseDotNetV2/package.json]: is this being used by any other task? Do we need to look at OSS governance?

Comment (bishal-pdMSFT) [reply Tasks/UseDotNetV2/package.json]: Why only @types

File: Tasks/UseDotNetV2/versioninstaller.ts [hunk 2]
```diff
@@ -43,7 +44,10 @@ export class VersionInstaller {
             // Extract
             console.log(tl.loc("ExtractingPackage", downloadPath));
             try {
-                var extPath = tl.osType().match(/^Win/) ? await toolLib.extractZip(downloadPath) : await toolLib.extractTar(downloadPath);
+                let tempDirectory = tl.getVariable('Agent.TempDirectory');
+                var destination: string = uuidV4();
+                destination = path.join( tempDirectory, destination.substr(destination.length - 4));
```
Comment (bishal-pdMSFT) [inline Tasks/UseDotNetV2/versioninstaller.ts]: use tinyguid logic from our UX

Comment (anuragc617) [reply Tasks/UseDotNetV2/versioninstaller.ts]: Done

File: Tasks/UseDotNetV2/package.json (hunk 1 above)
Comment (anuragc617) [reply Tasks/UseDotNetV2/package.json]: not using this now.

File: Tasks/UseDotNetV2/versioninstaller.ts [hunk 3]
```diff
@@ -43,7 +44,9 @@ export class VersionInstaller {
             // Extract
             console.log(tl.loc("ExtractingPackage", downloadPath));
             try {
-                var extPath = tl.osType().match(/^Win/) ? await toolLib.extractZip(downloadPath) : await toolLib.extractTar(downloadPath);
+                let tempDirectory = tl.getVariable('Agent.TempDirectory');
+                var destination = path.join( tempDirectory, this._tinyGuid());
+                var extPath = tl.osType().match(/^Win/) ? await toolLib.extractZip(downloadPath, destination) : await toolLib.extractTar(downloadPath);
```
Comment (hiyadav) [inline Tasks/UseDotNetV2/versioninstaller.ts]: Do we also need to update for linux os ?

File: Tasks/UseDotNetV2/versioninstaller.ts [hunk 4]
```diff
@@ -8,6 +8,7 @@ import * as toolLib from 'azure-pipelines-tool-lib/tool';
 
 import * as utils from "./versionutilities";
 import { VersionInfo } from "./models"
+const uuidV4 = require('uuid/v4');
```
Comment (hiyadav) [inline Tasks/UseDotNetV2/versioninstaller.ts]: you can remove this dependency

File: Tasks/PackerBuildV1/src/taskParameters.ts [hunk 5]
```diff
@@ -126,6 +126,14 @@ export default class TaskParameters {
         return inputPath;
     }
 
+    private _tinyGuid() {
+        return "yxxx".replace(/[xy]/g, function (c) {
+            var r = Math.random() * 16 | 0;
+            var v = c === "x" ? r : (r & 0x3 | 0x8);
+            return v.toString(16);
+        });
```
Comment (hiyadav) [inline Tasks/PackerBuildV1/src/taskParameters.ts]: Where is this getting consumed ?

File: Tasks/UseDotNetV2/versioninstaller.ts [hunk 6]
```diff
@@ -170,6 +173,14 @@ export class VersionInstaller {
         throw tl.loc("FileNameNotCorrectCompleteFileName", name);
     }
 
+    private _tinyGuid() {
+        return "yxxx".replace(/[xy]/g, function (c) {
+            var r = Math.random() * 16 | 0;
+            var v = c === "x" ? r : (r & 0x3 | 0x8);
```
Comment (hiyadav) [inline Tasks/UseDotNetV2/versioninstaller.ts]: Can you port this to other dotnetcore installer task versions as well

File: Tasks/UseDotNetV2/versioninstaller.ts (hunk 3 above)
Comment (hiyadav) [reply Tasks/UseDotNetV2/versioninstaller.ts]: Can you verify that the same long path works on linux, if not then we should update for linux as well

File: Tasks/UseDotNetV2/versioninstaller.ts (hunk 6 above)
Comment (vineetmimrot) [reply Tasks/UseDotNetV2/versioninstaller.ts]: Better move to common library.

File: Tasks/UseDotNetV2/versioninstaller.ts (hunk 4 above)
Comment (anuragc617) [reply Tasks/UseDotNetV2/versioninstaller.ts]: removed it

File: Tasks/UseDotNetV2/versioninstaller.ts (hunk 3 above)
Comment (anuragc617) [reply Tasks/UseDotNetV2/versioninstaller.ts]: verified, its working fine on linux agent

File: Tasks/PackerBuildV1/src/taskParameters.ts (hunk 5 above)
Comment (anuragc617) [reply Tasks/PackerBuildV1/src/taskParameters.ts]: added by mistake removed it.

File: Tasks/Common/utility-common-v2/tinyGuidUtility.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+export function tinyGuid() {
```
Comment (hiyadav) [inline Tasks/Common/utility-common-v2/tinyGuidUtility.ts]: you would need to upgrade the version of common utility, and also each task that utilizes this common utility.

File: Tasks/Common/azure-arm-rest-v2/package-lock.json [hunk 8]
```diff
@@ -25,11 +25,11 @@
       "integrity": "sha512-PR8oap9z2j+o455W3PwAfB4SX1p4GdJc9OHQaQV0V+iQS1IBY6dVgcNSQMkHAXb0V1bbuLOFBLanXPe5eSgGTQ==",
       "requires": {
         "minimatch": "3.0.4",
-        "mockery": "1.7.0",
-        "q": "1.4.1",
-        "semver": "5.5.0",
-        "shelljs": "0.3.0",
-        "uuid": "3.2.1"
+        "mockery": "^1.7.0",
```
Comment (bishal-pdMSFT) [inline Tasks/Common/azure-arm-rest-v2/package-lock.json]: I don't think you are adding any new package dependency. You can unstage these package.json changes.

File: Tasks/UseDotNetV2/make.json [hunk 9]
```diff
@@ -4,12 +4,27 @@
...
         {
             "source": "externals",
             "options": "-R"
         }
+    ],
+    "rm": [
+        {
+            "items": [
+                "node_modules/utility-common-v2/node_modules/azure-pipelines-task-lib",
+				"node_modules/docker-common-v2/node_modules/azure-pipelines-task-lib"
```
Comment (bishal-pdMSFT) [inline Tasks/UseDotNetV2/make.json]: fix spacing. Do not use tabs. there is a setting in vs code which automatically converts tabs to spaces

Comment (bishal-pdMSFT) [inline Tasks/UseDotNetV2/make.json]: Why this line? Is docker-common-v2 a dependency?

File: Tasks/UseDotNetV2/versioninstaller.ts [hunk 10]
```diff
@@ -43,7 +44,9 @@ export class VersionInstaller {
             // Extract
             console.log(tl.loc("ExtractingPackage", downloadPath));
             try {
-                var extPath = tl.osType().match(/^Win/) ? await toolLib.extractZip(downloadPath) : await toolLib.extractTar(downloadPath);
+                let tempDirectory = tl.getVariable('Agent.TempDirectory');
+                var destination = path.join( tempDirectory, tinyGuid());
```
Comment (bishal-pdMSFT) [inline Tasks/UseDotNetV2/versioninstaller.ts]: call it "extDirectory"?

File: Tasks/UseDotNetV2/versioninstaller.ts [hunk 11]
```diff
@@ -43,7 +44,9 @@ export class VersionInstaller {
             // Extract
             console.log(tl.loc("ExtractingPackage", downloadPath));
             try {
-                var extPath = tl.osType().match(/^Win/) ? await toolLib.extractZip(downloadPath) : await toolLib.extractTar(downloadPath);
+                let tempDirectory = tl.getVariable('Agent.TempDirectory');
```
[the remark this answers is not in the document]
Comment (anuragc617) [reply Tasks/UseDotNetV2/versioninstaller.ts]: older versions of task are using vsts-task-tool-lib/tool which doesn't support providing the extract directory path.

File: Tasks/UseDotNetV2/versioninstaller.ts (hunk 10 above)
Comment (anuragc617) [reply Tasks/UseDotNetV2/versioninstaller.ts]: done

File: Tasks/UseDotNetV2/make.json (hunk 9 above)
Comment (anuragc617) [reply Tasks/UseDotNetV2/make.json]: done

Comment (anuragc617) [reply Tasks/UseDotNetV2/make.json]: no dependency on docker-common-v2, removed it.

File: Tasks/Common/utility-common-v2/tinyGuidUtility.ts (hunk 7 above)
Comment (anuragc617) [reply Tasks/Common/utility-common-v2/tinyGuidUtility.ts]: done

**Pull request merged**